perf-test/components/SkottiePlayer/index.tsx

```diff
@@ -17,8 +17,11 @@ interface Props {
 }
 
 export default function Skottie ({ lottieURL, width, height }: Props) {
+  const canvasRef = useRef<HTMLCanvasElement>(null);
   const [id, setId] = useState<string>('');
+  let [ratio, setRatio] = useState<number>(1.0);
```

Comment:
setRatio is necessary? can't find any purpose this.

Reply:
This is also used in condition for avoiding unnecessary code run.
```js
// if DPR is same with cached value, don't run `canvasKit.MakeSWCanvasSurface`
if (dpr !== ratio && !changingRatio) {
```

Because `MakeSWCanvasSurface` is pretty expensive to run. It depends frame rate of device, if our device supports 120FPS, this function will be called 120 in a second.